### PR TITLE
add meson build system

### PR DESCRIPTION
**Pull request by 1ace**
https://mesonbuild.com/Quick-guide.html for those unfamiliar with Meson.

There's 2 kinds of options now:
- `windows` which can be one of `all`, `wayland`, or `x11`
- the backend are now all auto-detected by default (ie. enabled if their dependencies are met); you can enable all of them with `-D auto_features=enabled`, or one at a time with `-D freeimage=enabled`

### CODE REVIEW

File: meson_get_version.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env python3
+# encoding=utf-8
+
+import os
+import subprocess
+import sys
+
+try:
+    sys.stdout.write(subprocess.check_output([
```
Comment (emersion) [inline meson_get_version.py]: This can be done without Python. See e.g. https://github.com/swaywm/swaylock/blob/master/meson.build#L53

Comment (1ace) [reply meson_get_version.py]: haha hello, didn't expect to see you here :)

your solution works to set the define for the C files, but not for the project version; I guess I can simplify the whole thing for now, and if someone decides later that we need the project verion to be accurate as well we can change this.

**Comment by emersion** [comment]
Note: instead of pushing repeatedly, with a sr.ht account you can copy-paste any build manifest [on the submit page](https://builds.sr.ht/submit) and SSH into failed builds to investigate.

**Comment by 1ace** [comment]
ah, good to know, thanks :)

it did seem like a waste to re-run all the builds when only one of them needed to run

**Comment by 1ace** [comment]
OK, I think everything's ready to be merged now :)

**Comment by 1ace** [comment]
added one more commit to drop the old Makefile and update the docs

**Comment by 1ace** [comment]
BTW the reason I did this is that I wanted to look at an imv bug and when I ran `make clean` it nuked my /var/tmp, which prompted me to look at the Makefile and noticed it was a fragile mess, so I decided to write a clean meson.build instead, with all the advantages of meson over a plain makefile as well.

**Comment by eXeC64** [comment]
I was on the fence about this for a while - but when I reinstalled linux and the build failed without helpfully telling me what was missing, I figured this was a worthwhile improvement.

Thanks for your work, I'm applying your patch to master.

**Pull request closed without merging**